js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+        } else if (exchangeType === 'futures') {
+            // JUST FOR DEMONSTRATIONAL PURPOSES
+            // const response = await this.futuresPublicGetMarkets (params);
+            // i.e. : data = this.safeValue (response, 'result', []);
+        }
+        const result = [];
+        for (let i = 0; i < data.length; i++) {
+            const market = data[i];
+            const id = this.safeString (market, 'name');
+            // const [ baseId, quoteId ] = id.split ('_');  // We do have them in properties, so, I don't think this is needed ?
+            const baseId = this.safeString (market, 'stock');
+            const quoteId = this.safeString (market, 'money');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const priceScale = this.safeInteger (market, 'moneyPrec'); // <<< TODO >>>  strings 1e-8
+            const quantityScale = this.safeInteger (market, 'stockPrec');
+            const pricePrecision = 1 / Math.pow (10, priceScale);
+            const quantityPrecision = 1 / Math.pow (10, quantityScale);
+            const minAmount = this.safeNumber (market, 'minAmount');
+            const type = 'spot';
+            result.push (this.safeMarketObject ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'type': type,
+                'spot': exchangeType === 'spot',
+                'margin': undefined,
+                'active': undefined,
+                'taker': undefined,
+                'maker': undefined,
+                'precision': {
+                    'price': pricePrecision,
+                    'amount': quantityPrecision,
+                },
+                'limits': {
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'amount': {
+                        'min': minAmount,
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': this.safeNumber (market, 'feePrec'),     // <<< TODO >>> I am not yet sure this is correct.
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            }));
+        }
+        return result;
+    }
+
+    async fetchTicker (symbol = undefined, params = {}) {
+        const [exchangeType, request] = this.checkExchangeType ('fetchTicker', params);
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        request['market'] = market['id'];
+        let ticker = {};
```

Comment:
We should set it to `undefined` by default to avoid instantiating empty objects, if possible.

Reply:
done